src/sentry/relay/config/metric_extraction.py

```diff
@@ -529,32 +529,35 @@ def _convert_aggregate_and_query_to_metrics(
         metric_specs_and_hashes = []
         # Create as many specs as we support
         for spec_version in OnDemandMetricSpecVersioning.get_spec_versions():
-            on_demand_spec = OnDemandMetricSpec(
-                field=aggregate,
-                query=query,
-                environment=environment,
-                groupbys=groupbys,
-                spec_type=spec_type,
-                spec_version=spec_version,
-            )
-            metric_spec = on_demand_spec.to_metric_spec(project)
-            # TODO: switch to validate_rule_condition
-            if (condition := metric_spec.get("condition")) is not None:
-                validate_sampling_condition(json.dumps(condition))
-            else:
-                metrics.incr(
-                    "on_demand_metrics.missing_condition_spec",
-                    tags={"prefilling": prefilling},
+            try:
```

Comment:
I added an extra try block since the Unicode error would prevent generating specs for that project/org.